### PR TITLE
fix broken compile

### PR DESCRIPTION
**Pull request by psychocoderHPC**
update usage of `::alpaka::workdiv::WorkDivMembers` (interfaces were changed in alpaka)

### CODE REVIEW

File: src/libPMacc/include/eventSystem/tasks/TaskSetCurrentSizeOnDevice.hpp [hunk 1]
```diff
@@ -97,7 +97,8 @@ class TaskSetCurrentSizeOnDevice : public StreamTask
         kernelSetValueOnDeviceMemory kernel;
         alpaka::workdiv::WorkDivMembers<alpaka::dim::DimInt<1u>, AlpakaIdxSize> workDiv(
             static_cast<AlpakaIdxSize>(1u),
-            static_cast<AlpakaIdxSize>(1u));
+            static_cast<AlpakaIdxSize>(1u),
+            1);
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/tasks/TaskSetCurrentSizeOnDevice.hpp]: I think we should make a `static_cast<AlpakaIdxSize>(1u)` here too.

File: src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp [hunk 2]
```diff
@@ -201,7 +201,8 @@ class TaskSetValue<T_ValueType, T_dim, true> : public TaskSetValueBase<T_ValueTy
         kernelSetValue kernel;
         alpaka::workdiv::WorkDivMembers<alpaka::dim::DimInt<dim>, AlpakaIdxSize> workDiv(
             gridSize,
-            blockSize);
+            blockSize,
+            1);
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp]: I think we should make a `static_cast<AlpakaIdxSize>(1u)` here.

File: src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp [hunk 3]
```diff
@@ -273,7 +274,8 @@ class TaskSetValue<T_ValueType, T_dim, false> : public TaskSetValueBase<T_ValueT
...
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp]: Okay static_cast.

File: src/picongpu/include/simulation_classTypes.hpp [hunk 4]
```diff
@@ -72,5 +72,5 @@ namespace picongpu
         PMACC_KERNEL_CATCH(::alpaka::wait::wait(::PMacc::Environment<>::get().DeviceManager().getAccDevice()), "picKernelArea: crash before kernel call");\
         ::PMacc::AreaMapping<area, MappingDesc> mapper(description);\
         ::PMacc::TaskKernel * const taskKernel(::PMacc::Environment<>::get().Factory().createTaskKernel(#KERNEL));\
-        auto const exec(::alpaka::exec::create<::PMacc::AlpakaAcc<DIM>>(::alpaka::workdiv::WorkDivMembers<DIM, AlpakaIdxSize>(mapper.getGridDim(), block), KERNEL\
+        auto const exec(::alpaka::exec::create<::PMacc::AlpakaAcc<DIM>>(::alpaka::workdiv::WorkDivMembers<DIM, AlpakaIdxSize>(mapper.getGridDim(),block,1), KERNEL\
```
Comment (erikzenker) [inline src/picongpu/include/simulation_classTypes.hpp]: Cast :ghost:

File: src/libPMacc/include/eventSystem/events/kernelEvents.hpp [hunk 5]
```diff
@@ -86,5 +86,5 @@
     {\
         PMACC_KERNEL_CATCH(::alpaka::wait::wait(::PMacc::Environment<>::get().DeviceManager().getAccDevice()), "__cudaKernel: crash before kernel call");\
         ::PMacc::TaskKernel * const taskKernel(::PMacc::Environment<>::get().Factory().createTaskKernel(#KERNEL));\
-        auto const exec(::alpaka::exec::create<::PMacc::AlpakaAcc<DIM>>(::alpaka::workdiv::WorkDivMembers<DIM, ::PMacc::AlpakaIdxSize>(__VA_ARGS__), KERNEL\
+        auto const exec(::alpaka::exec::create<::PMacc::AlpakaAcc<DIM>>(::alpaka::workdiv::WorkDivMembers<DIM, ::PMacc::AlpakaIdxSize>(__VA_ARGS__,1), KERNEL\
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/kernelEvents.hpp]: I think we should make a `static_cast<AlpakaIdxSize>(1u)` here.

**Comment by erikzenker** [comment]
Where is AlpakaIdxSize defined ?

**Comment by psychocoderHPC** [comment]
In `libPMacc` in `types.h`

**Comment by erikzenker** [comment]
Can we add this information to the included headers ?

`#include "types.h" /* AlpakaIdxSize */`

Otherwise its hard to find defines in a non ide supported editor.

**Comment by psychocoderHPC** [comment]
branch updated!

`#include "types.h" /* AlpakaIdxSize */` might be helps with `AlpakaIdxSize` but there are also other important defines which are not in the comment.

**Comment by erikzenker** [comment]
Nicely done, ready to merge! :+1:

**Pull request merged**